conan/tools/cmake/cmake.py

```diff
@@ -56,8 +56,11 @@ def __init__(self, conanfile):
         configure_preset = get_configure_preset(cmake_presets, conanfile)
 
         self._generator = configure_preset["generator"]
-        self._toolchain_file = configure_preset["toolchainFile"]
         self._cache_variables = configure_preset["cacheVariables"]
+        if "toolchainFile" in configure_preset:
+            self._toolchain_file = configure_preset["toolchainFile"]
+        else:
+            self._toolchain_file = self._cache_variables["CMAKE_TOOLCHAIN_FILE"]
```

Comment:
It is not needed to manage the cache variable it will be applied later automatically when doing `arg_list.extend(['-D{}="{}"'.format(k, v) for k, v in self._cache_variables.items()])`

```suggestion
       self._toolchain_file = configure_preset.get("toolchainFile")
```

Reply:
> It is not needed to manage the cache variable

Yes, you right. I just didn't thought about it.  So final patch is much simplier.